src/get_subid_owners.c

```diff
@@ -18,20 +23,24 @@ static void usage(void)
 
 int main(int argc, char *argv[])
 {
-	int i, n;
-	uid_t *uids;
+	int    i, n;
+	long   l;
+	uid_t  *uids;
 
 	log_set_progname(Prog);
 	log_set_logfd(stderr);
 	if (argc < 2) {
 		usage();
 	}
-	if (argc == 3 && strcmp(argv[1], "-g") == 0)
-		n = subid_get_gid_owners(atoi(argv[2]), &uids);
-	else if (argc == 2 && strcmp(argv[1], "-h") == 0)
+	if (argc == 3 && strcmp(argv[1], "-g") == 0) {
+		str2sl(&l, argv[2]);
+		n = subid_get_gid_owners(l, &uids);
+	} else if (argc == 2 && strcmp(argv[1], "-h") == 0) {
 		usage();
-	else
-		n = subid_get_uid_owners(atoi(argv[1]), &uids);
+	} else {
+		str2sl(&l, argv[1]);
+		n = subid_get_uid_owners(l, &uids);
```

Comment:
Hm, you're obviously not introducing this, but technically I guess l should be unsigned long?  Going by 'typedef __kernel_uid32_t        uid_t;' in the kernel.

Reply:
We can use `uid_t`.  That is:

```c
str2i(uid_t, &u, argv[1]);
```